js/layout-javascript/agenda-code.js

```diff
@@ -2550,7 +2561,27 @@ DynamicList.prototype.addDetailViewData = function(entry) {
       content = entry.originalData[dynamicDataObj.column];
     }
 
-    content = _this.Utils.String.toFormattedString(content);
+    if (dynamicDataObj.type === 'image') {
+      var imagesContentData = _this.Utils.Record.imageContent(entry.originalData[dynamicDataObj.column]);
```

Comment:
The logic for everything in this `if` clause is still repeated. They look pretty similar, if not exactly the same in all files. This is something where if there's bug or enhancement needed, it'll become very complex to update and consider backward compatibility. It's also not something that users would really need to customise, and therefore has no reason to be repeated in the layout JS files.